bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+
+  // This is a bit hacked, as it requires invoking select for each of the substrategies, which are structured to
+  // pick the best assignment for the current position angle constraint. This, of course, might result in different
+  // position angles for substrategies if we naively call select on each, so instead, we hack the context to
+  // handle each position angle constraint and then pick the best one.
+  override def select(ctx: ObsContext, mt: MagnitudeTable): Future[Option[AgsStrategy.Selection]] = {
+    // We hack the obs context to be a list of obs contexts with all possible position angles, and with fixed
+    // position angle constraint so that guide probes will not be allowed to take on different angles in select calls.
+    def ctxToFixedList: List[ObsContext] = {
+      val ctxFixed = {
+        val instFixed = ctx.getInstrument match {
+          case pac : SPInstObsComp with PosAngleConstraintAware =>
+            val newPac = pac.clone.asInstanceOf[SPInstObsComp with PosAngleConstraintAware]
+            newPac.setPosAngleConstraint(PosAngleConstraint.FIXED)
+            newPac
+          case oth : SPInstObsComp => oth
+        }
+        ctx.withInstrument(instFixed)
+      }
+      (ctx.getPosAngleConstraint match {
+        case PosAngleConstraint.FIXED                                            => List(0)
+        case PosAngleConstraint.FIXED_180 | PosAngleConstraint.PARALLACTIC_ANGLE => List(0, 180)
+        case PosAngleConstraint.UNBOUNDED                                        => Range(0, 360, 15).toList
+      }).map(a => ctxFixed.withPositionAngle(ctx.getPositionAngle.add(a, Angle.Unit.DEGREES)))
+    }
+
+    // For each ObsContext, perform the selection and return it as a future.
+    val noStrategy: Option[AgsStrategy.Selection] = None
+    def ctxSelect(cCtx: ObsContext): Future[(ObsContext, Option[AgsStrategy.Selection])] =
+      Future.fold(strategies.map(_.select(cCtx, mt)))(cCtx -> noStrategy) {
+        case ((_, resOpt), curOpt) =>
+          cCtx -> curOpt.map(cur => AgsStrategy.Selection(cur.posAngle, resOpt.fold(cur.assignments)(cur.assignments ++ _.assignments)))
+      }
+
+    // We need a way to compare AGS selections to determine which is the best across the list of generated
+    // ObsContexts. We assign a 4-tuple to each selection in such a way that the minimum of the 4-tuples represents
+    // the best selection.
+    // 1. Quality index is the quality converted to an int, and then summed over all targets.
+    // 2. Vignetting index is vignetting for each vignetting guide probe, summed across all.
+    // 3. Best quality across all targets. Used because for magnitude, we only want best quality class in comparison.
+    // 4. Magnitude is magnitude of the brightest star in the assignment in the best quality class.
+    def selValue(cCtx: ObsContext, sel: AgsStrategy.Selection): (Int, Double, Int, Double) = {
```

Comment:
Perhaps instead of a 4-tuple you can make a small case class that would be more explicative